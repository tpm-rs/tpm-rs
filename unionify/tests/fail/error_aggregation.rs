use tpm2_rs_unionify::UnionSize;

#[derive(UnionSize)]
pub enum Foo {
    A { a1: u8, a2: u8 },
    B { b1: u64, b2: u64 },
    C { c1: u8, c2: u16 },
}

fn main() {}
