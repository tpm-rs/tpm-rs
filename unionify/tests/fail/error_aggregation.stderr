error: UnionSize is not derivable for enum variants containing more than one field
 --> tests/fail/error_aggregation.rs:5:7
  |
5 |     A { a1: u8, a2: u8 },
  |       ^^^^^^^^^^^^^^^^^^

error: UnionSize is not derivable for enum variants containing more than one field
 --> tests/fail/error_aggregation.rs:6:7
  |
6 |     B { b1: u64, b2: u64 },
  |       ^^^^^^^^^^^^^^^^^^^^

error: UnionSize is not derivable for enum variants containing more than one field
 --> tests/fail/error_aggregation.rs:7:7
  |
7 |     C { c1: u8, c2: u16 },
  |       ^^^^^^^^^^^^^^^^^^^
